src/lib/y2network/routing_helpers.rb

```diff
@@ -0,0 +1,110 @@
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2network/interface"
+require "y2network/routing_table"
+require "y2network/route"
+
+module Y2Network
+  module RoutingHelpers
+
+  private
+
+    MISSING_VALUE = "-".freeze
+    private_constant :MISSING_VALUE
+    DEFAULT_DEST = "default".freeze
+    private_constant :DEFAULT_DEST
+
+    # Find configured network interfaces
+    #
+    # Configured interfaces has a configuration (ifcfg file) assigned.
+    #
+    # @return [Array<Interface>] Detected interfaces
+    # @see Yast::NetworkInterfaces.Read
+    def find_interfaces
+      Yast::NetworkInterfaces.Read
+      # TODO: for the time being, we are just relying in the underlying stuff.
+      Yast::NetworkInterfaces.List("").map do |name|
+        Y2Network::Interface.new(name)
+      end
+    end
+
+    # Given an IP and a netmask, returns a valid IPAddr object
+    #
+    # @param ip_str      [String] IP address; {MISSING_VALUE} means that the IP is not defined
+    # @param netmask_str [String] Netmask; {MISSING_VALUE} means that no netmask was specified
+    # @return [IPAddr,nil] The IP address or `nil` if the IP is missing
+    def build_ip(ip_str, netmask_str = MISSING_VALUE)
+      return nil if ip_str == MISSING_VALUE
+
+      ip = IPAddr.new(ip_str)
+      netmask_str == MISSING_VALUE ? ip : ip.mask(netmask_str)
+    end
+
+    # Build a route given a hash from the SCR agent
+    #
+    # @param hash [Hash] Hash from the `.routes` SCR agent
+    # @return Route
+    def build_route(hash)
```

Comment:
something for `Y2Network::Route` ?

Reply:
MM maybe, but then it should be something like build_from() ... I think will just let the current methods in the original readers